pkg/didcomm/protocol/route/service.go

```diff
@@ -194,14 +216,20 @@ func (s *Service) handleRequest(msg *service.DIDCommMsg, myDID, theirDID string)
 
 func (s *Service) handleGrant(msg *service.DIDCommMsg) error {
 	// unmarshal the payload
-	grant := &Grant{}
+	grantMsg := &Grant{}
 
-	err := json.Unmarshal(msg.Payload, grant)
+	err := json.Unmarshal(msg.Payload, grantMsg)
 	if err != nil {
 		return fmt.Errorf("route grant message unmarshal : %w", err)
 	}
 
-	// TODO https://github.com/hyperledger/aries-framework-go/issues/948 integrate with framework components
+	// check if there are any channels registered for the message ID
+	grantCh := s.getRouteRegistrationCh(grantMsg.ID)
+
+	if grantCh != nil {
```

Comment:
when does nil happen?

Reply:
If the channel is not registered for the msgID. basically when the agent doesn't recognize the message ID.